tv/tv.cpp

```diff
@@ -568,10 +574,15 @@ const llvm::Module * unwrapModule(llvm::Any IR) {
   llvm_unreachable("Unknown IR unit");
 }
 
+
 const char* skip_pass_list[] = {
```

Comment:
can you check if there's an API in LLVM that allow us to query if it's a module pass? It would be nice to get rid of this list.

Reply:
PassBuilder doesn't seem to have any way to retrieving data about registered passes. Might be something we wanna add going fwd

Reply:
It seems the `llvm::Any IR` argument that is defined below describes the level of the pass.
I found this from PassInstrumentation.h:
```
  /// AfterPass instrumentation point - takes \p Pass instance that has
  /// just been executed and constant reference to \p IR it operates on.
  /// \p IR is guaranteed to be valid at this point.
  template <typename IRUnitT, typename PassT>
  void runAfterPass(const PassT &Pass, const IRUnitT &IR,
                    const PreservedAnalyses &PA) const {

```
But would it be slightly loose to depend on the contents of IR? LLVM may pass an incorrect `IR`.

Reply:
(to clarify: `llvm::Any IR` is at line 729 of tv.cpp)

Reply:
yeah, I'm not seeing a safe way to do it with the current APIs.

Reply:
I added an issue for this: #617